opencv: Undefined symbols for architecture x86_64:   "Imf::Header::Header

While compiling an oF app using ofxOpenCv I got:

```
Undefined symbols for architecture x86_64:
  "Imf::Header::Header(int, int, float, Imath::Vec2<float> const&, float, Imf::LineOrder, Imf::Compression)", referenced from:
      cv::ExrEncoder::write(cv::Mat const&, std::__1::vector<int, std::__1::allocator<int> > const&) in opencv.a(grfmt_exr.cpp.o)
  "Imf::Header::channels()", referenced from:
      cv::ExrEncoder::write(cv::Mat const&, std::__1::vector<int, std::__1::allocator<int> > const&) in opencv.a(grfmt_exr.cpp.o)
  "Imf::Channel::Channel(Imf::PixelType, int, int, bool)", referenced from:
      cv::ExrEncoder::write(cv::Mat const&, std::__1::vector<int, std::__1::allocator<int> > const&) in opencv.a(grfmt_exr.cpp.o)
  "Imf::OutputFile::setFrameBuffer(Imf::FrameBuffer const&)", referenced from:
      cv::ExrEncoder::write(cv::Mat const&, std::__1::vector<int, std::__1::allocator<int> > const&) in opencv.a(grfmt_exr.cpp.o)
  "Imf::OutputFile::writePixels(int)", referenced from:
      cv::ExrEncoder::write(cv::Mat const&, std::__1::vector<int, std::__1::allocator<int> > const&) in opencv.a(grfmt_exr.cpp.o)
  "Imf::OutputFile::~OutputFile()", referenced from:
      cv::ExrEncoder::write(cv::Mat const&, std::__1::vector<int, std::__1::allocator<int> > const&) in opencv.a(grfmt_exr.cpp.o)
  "Imf::FrameBuffer::insert(char const*, Imf::Slice const&)", referenced from:
      cv::ExrDecoder::readData(cv::Mat&) in opencv.a(grfmt_exr.cpp.o)
      cv::ExrEncoder::write(cv::Mat const&, std::__1::vector<int, std::__1::allocator<int> > const&) in opencv.a(grfmt_exr.cpp.o)
  "Imf::InputFile::setFrameBuffer(Imf::FrameBuffer const&)", referenced from:
      cv::ExrDecoder::readData(cv::Mat&) in opencv.a(grfmt_exr.cpp.o)
  "Imf::InputFile::readPixels(int, int)", referenced from:
      cv::ExrDecoder::readData(cv::Mat&) in opencv.a(grfmt_exr.cpp.o)
  "Imf::ChannelList::insert(char const*, Imf::Channel const&)", referenced from:
      cv::ExrEncoder::write(cv::Mat const&, std::__1::vector<int, std::__1::allocator<int> > const&) in opencv.a(grfmt_exr.cpp.o)
  "Imf::globalThreadCount()", referenced from:
      cv::ExrDecoder::readHeader() in opencv.a(grfmt_exr.cpp.o)
      cv::ExrEncoder::write(cv::Mat const&, std::__1::vector<int, std::__1::allocator<int> > const&) in opencv.a(grfmt_exr.cpp.o)
  "Imf::InputFile::InputFile(char const*, int)", referenced from:
      cv::ExrDecoder::readHeader() in opencv.a(grfmt_exr.cpp.o)
  "Imf::Slice::Slice(Imf::PixelType, char*, unsigned long, unsigned long, int, int, double, bool, bool)", referenced from:
      cv::ExrDecoder::readData(cv::Mat&) in opencv.a(grfmt_exr.cpp.o)
      cv::ExrEncoder::write(cv::Mat const&, std::__1::vector<int, std::__1::allocator<int> > const&) in opencv.a(grfmt_exr.cpp.o)
  "Imf::hasChromaticities(Imf::Header const&)", referenced from:
      cv::ExrDecoder::readHeader() in opencv.a(grfmt_exr.cpp.o)
  "Imf::Header::~Header()", referenced from:
      cv::ExrEncoder::write(cv::Mat const&, std::__1::vector<int, std::__1::allocator<int> > const&) in opencv.a(grfmt_exr.cpp.o)
  "Imf::chromaticities(Imf::Header const&)", referenced from:
      cv::ExrDecoder::readHeader() in opencv.a(grfmt_exr.cpp.o)
  "Imf::Header::dataWindow() const", referenced from:
      cv::ExrDecoder::readHeader() in opencv.a(grfmt_exr.cpp.o)
  "Imf::Chromaticities::Chromaticities(Imath::Vec2<float> const&, Imath::Vec2<float> const&, Imath::Vec2<float> const&, Imath::Vec2<float> const&)", referenced from:
      cv::ExrDecoder::ExrDecoder() in opencv.a(grfmt_exr.cpp.o)
  "Imf::OutputFile::OutputFile(char const*, Imf::Header const&, int)", referenced from:
      cv::ExrEncoder::write(cv::Mat const&, std::__1::vector<int, std::__1::allocator<int> > const&) in opencv.a(grfmt_exr.cpp.o)
  "Imf::Header::channels() const", referenced from:
      cv::ExrDecoder::readHeader() in opencv.a(grfmt_exr.cpp.o)
  "Imf::InputFile::header() const", referenced from:
      cv::ExrDecoder::readHeader() in opencv.a(grfmt_exr.cpp.o)
  "Imf::ChannelList::findChannel(char const*) const", referenced from:
      cv::ExrDecoder::readHeader() in opencv.a(grfmt_exr.cpp.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

I have encountered this problem in my personal project, though I'm not sure what exactly causes this error (undefined symbols are referred in the opencv static library). 

From the above error messages, there are missing symbols related to openexr.  After digging into the problem a bit, there seems to be two workaround:

1. add cmake option `-DWITH_OPENEXR=OFF` to the build script for opencv and rebuild opencv.a
2. generate makefile with `-DWITH_OPENEXR=ON` and make libIlmImf.a adn add it to compiling source


I'm not still sure what is the best way to fix this, and the exact reason that I had this error. Is there anyone there who has a similar problem?

--- a/addons/ofxOpenCv/scripts/formulas/opencv.sh
+++ b/addons/ofxOpenCv/scripts/formulas/opencv.sh
@@ -348,6 +348,7 @@ function build() {
       -DWITH_QUICKTIME=OFF \
       -DWITH_V4L=OFF \
       -DWITH_PVAPI=OFF \
+      -DWITH_OPENEXR=OFF \
       -DBUILD_TESTS=OFF \
       -DBUILD_PERF_TESTS=OFF >> "${LOG}" 2>&1
 

--- a/addons/ofxOpenCv/scripts/formulas/opencv.sh
+++ b/addons/ofxOpenCv/scripts/formulas/opencv.sh
@@ -95,6 +95,7 @@ function build() {
       -DWITH_QUICKTIME=OFF \
       -DWITH_V4L=OFF \
       -DWITH_PVAPI=OFF \
+      -DWITH_OPENEXR=OFF \
       -DBUILD_TESTS=OFF \
       -DBUILD_PERF_TESTS=OFF 2>&1 | tee -a ${LOG}
     echo "CMAKE Successful"

